### PR TITLE
[REVIEW] always test both legacy and per-thread default stream

### LINKED ISSUE #555: [BUG] RMM CI should not only test PTDS mode
Recently a decision was made to build and test cuDF with Per-Thread Default Stream set to ON, because this is a more conservative test than OFF and therefore it should be sufficient to test cuDF only with PTDS ON.

However, for some reason that change was also applied to RMM which is NOT a valid testing approach since unlike cuDF, RMM has separate code paths for PTDS=ON and PTDS=OFF. If we only test with PTDS ON, then we will not be testing both paths.

Instead, librmm includes CMake code to compile some of its tests in both modes, creating separate PTDS and non-PTDS test binaries.  So for C++ / gtesting, PTDS should not be globally configured for RMM. 

For Python testing, we need to change to building the Cython bindings twice (once with PTDS=ON and once with PTDS=OFF) and run the pytests twice.

### PR DESCRIPTION
**Pull request by rongou**
Fixes #555 

@harrism

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

**Comment by harrism** [comment]
Also need to change the CI scripts.

File: build.sh [hunk 1]
```diff
@@ -18,16 +18,15 @@ ARGS=$*
 # script, and that this script resides in the repo dir!
 REPODIR=$(cd $(dirname $0); pwd)
 
-VALIDARGS="clean librmm rmm -v -g -n -s --ptds -h"
-HELP="$0 [clean] [librmm] [rmm] [-v] [-g] [-n] [-s] [--ptds] [-h]
+VALIDARGS="clean librmm rmm -v -g -n -s -h"
```
Comment (harrism) [inline build.sh:21]: I don't think you want to remove the option to build --ptds in this file, do you? You just want to remove the use of it in CI (the other file change you made). Somebody may want to use this to build the benchmarks with PTDS, for example?

**Review by harrism — CHANGES_REQUESTED** [review]

File: build.sh (hunk 1 above)
Comment (rongou) [reply build.sh:21]: As far as I can tell this script doesn't allow you to build the benchmarks. The option `BUILD_BENCHMARKS` is set to `OFF` in the cmake file, and there is currently no way to turn it on through this script.

Comment (harrism) [reply build.sh:21]: OK. But if we fix that, all of these changes will have to be put back?

Comment (rongou) [reply build.sh:21]: I suppose so. But as it stands now, passing in `--ptds` has no effect, so it's kinda misleading.

**Review by harrism — APPROVED** [review]

File: build.sh (hunk 1 above)
Comment (kkraus14) [reply build.sh:21]: We should add the ability to build benchmarks to this script, could you leave the option in even if it has no effect for now?

**Review by kkraus14 — CHANGES_REQUESTED** [review]
Implementation looks good. Would just like the `build.sh` script change reverted so we can properly add support to build the benchmarks to it.

File: build.sh (hunk 1 above)
Comment (rongou) [reply build.sh]: Done.

**Review by kkraus14 — APPROVED** [review]

**Pull request merged**